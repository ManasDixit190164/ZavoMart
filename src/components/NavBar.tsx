import React from "react";
import { View, TextInput, StyleSheet, Text, StatusBar } from "react-native";
import Icon from "react-native-vector-icons/Ionicons";
import configData from "../configs/appConfig.json";

interface NavBarProps {
  query: string;
  onChangeText: (text: string) => void;
}

export const NavBar: React.FC<NavBarProps> = ({ query, onChangeText }) => {
  return (
    <View style={styles.container}>
      <StatusBar />
      <View style={styles.logo}>
        <Icon name={configData.Homepage.navIconName} size={26} />
        <Text style={styles.logoText}>{configData.Homepage.navText}</Text>
      </View>
      <TextInput
        placeholder={configData.Homepage.searchPlaceHolderText}
        value={query}
        onChangeText={onChangeText}
        style={styles.input}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    margin: 10,
    backgroundColor: "#eee",
    borderRadius: 8,
  },
  input: {
    padding: 10,
    borderWidth: 1,
    borderRadius: 12,
    marginTop: 8,
  },
  logo: {
    flexDirection: "row",
    alignItems: "center",
  },
  logoText: { fontSize: 24, fontWeight: "500", paddingLeft: 10 },
});
