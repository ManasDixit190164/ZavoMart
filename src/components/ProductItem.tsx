import React, { useState } from "react";
import {
  View,
  Text,
  Image,
  StyleSheet,
  TouchableOpacity,
  Animated,
} from "react-native";
import { Product } from "../models/Product";

interface ProductItemProps {
  product: Product;
  onPress: () => void;
  onToggleFavorite: (id: number) => void;
}

export const ProductItem: React.FC<ProductItemProps> = ({
  product,
  onPress,
  onToggleFavorite,
}) => {
  const [scaleAnim] = useState(new Animated.Value(1));

  const onPressIn = () => {
    Animated.spring(scaleAnim, {
      toValue: 0.95,
      useNativeDriver: true,
    }).start();
  };

  const onPressOut = () => {
    Animated.spring(scaleAnim, {
      toValue: 1,
      useNativeDriver: true,
    }).start();
  };

  return (
    <TouchableOpacity
      style={styles.container}
      onPress={onPress}
      activeOpacity={1}
      onPressIn={onPressIn}
      onPressOut={onPressOut}
    >
      <Animated.View
        style={[styles.card, { transform: [{ scale: scaleAnim }] }]}
      >
        <Image
          source={{ uri: product.image }}
          style={styles.image}
          resizeMode="contain"
        />
        <View style={styles.info}>
          <Text style={styles.name}>{product.name}</Text>
          <Text style={styles.price}>${product.price}</Text>
        </View>
        <TouchableOpacity
          onPress={() => onToggleFavorite(product.id)}
          style={styles.favoriteButton}
        >
        </TouchableOpacity>
      </Animated.View>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  container: {
    alignItems: "center",
    paddingHorizontal: 8,
  },
  image: {
    width: 80,
    height: 80,
  },
  info: {
    flex: 1,
    marginLeft: 10,
  },
  name: {
    fontSize: 16,
    fontWeight: "bold",
  },
  price: {
    marginTop: 5,
    fontSize: 14,
    color: "#888",
  },
  favoriteButton: {
    padding: 5,
    borderRadius: 5,
  },
  favoriteText: {
    fontSize: 12,
  },
  card: {
    flexDirection: "row",
    padding: 10,
    borderWidth: 1,
    borderColor: "#ccc",
    alignItems: "center",
    marginVertical: 4,
    borderRadius: 8,
    width: "100%",
    backgroundColor: "#fff",
    shadowColor: "#000", 
    shadowOffset: { width: 0, height: 1 }, 
    shadowOpacity: 0.1,
    shadowRadius: 2,
    elevation: 2,
  },
});
