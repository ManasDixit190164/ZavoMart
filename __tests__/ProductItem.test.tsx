import { ProductItem } from '@/src/components/ProductItem';
import renderer from 'react-test-renderer';
import { Product } from '@/src/models/Product';

it(`renders correctly`, () => {
  const tree = renderer.create(<ProductItem onPress={()=>{}} onToggleFavorite={()=>{}} product={{id : 1 , description:"" , image:"" , name:"Jean" , price:1000 , isFavorite:false}} />).toJSON();

  expect(tree).toMatchSnapshot();
});
