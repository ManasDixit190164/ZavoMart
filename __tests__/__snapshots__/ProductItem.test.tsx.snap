// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "opacity": 1,
      "paddingHorizontal": 8,
    }
  }
>
  <View
    collapsable={false}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderColor": "#ccc",
        "borderRadius": 8,
        "borderWidth": 1,
        "elevation": 2,
        "flexDirection": "row",
        "marginVertical": 4,
        "padding": 10,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 2,
        "transform": [
          {
            "scale": 1,
          },
        ],
        "width": "100%",
      }
    }
  >
    <Image
      resizeMode="contain"
      source={
        {
          "uri": "",
        }
      }
      style={
        {
          "height": 80,
          "width": 80,
        }
      }
    />
    <View
      style={
        {
          "flex": 1,
          "marginLeft": 10,
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        Jean
      </Text>
      <Text
        style={
          {
            "color": "#888",
            "fontSize": 14,
            "marginTop": 5,
          }
        }
      >
        $
        1000
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "borderRadius": 5,
          "opacity": 1,
          "padding": 5,
        }
      }
    />
  </View>
</View>
`;
